### PR TITLE
Add method to get Notepad++ version

### PR DESCRIPTION
**Pull request by gandarez**
This PR adds a method to retrieve the full version from Notepad++ like ` 8.4.7`.

### CODE REVIEW

**Review by kbilsted — CHANGES_REQUESTED** [review]

File: Visual Studio Project Template C#/PluginInfrastructure/NotepadPPGateway.cs [hunk 1]
```diff
@@ -119,6 +120,27 @@ public void SetCurrentLanguage(LangType language)
...
+		/// Gets Notepad++ version.
+		/// </summary>
+		public string GetNppVersion()
+		{
+			var msgPtr = Win32.SendMessage(PluginBase.nppData._nppHandle, (uint)NppMsg.NPPM_GETNPPVERSION, 0, 0);
+
+			try 
+			{
+				var num = Convert.ToInt32(msgPtr.ToString());
+                var high = num >> 16;
```
[the remark this answers is not in the document]
Comment (gandarez) [reply Visual Studio Project Template C#/PluginInfrastructure/NotepadPPGateway.cs:134]: What exactly is weird? I didn't get you

**Comment by gandarez** [comment]
Any thoughts?

**Comment by gandarez** [comment]
Hello?

**Review by oSerenity — APPROVED** [review]

**Comment by molsonkiko** [comment]
IMO it would be better for a `GetNppVersion` method to return a 2-int array containing (major, minor*10 + bugfix) or a 3-int array containing (major, minor, bugfix).
Returning an int array is more useful because the user doesn't need to parse a string to get the major, minor, or bugfix version. 

```
/// <summary>
/// 3-int array: {major, minor, bugfix}<br></br>
/// Thus GetNppVersion() would return {8, 5, 0} for version 8.5.0
/// and {7, 7, 1} for version 7.7.1
/// </summary>
/// <returns></returns>
public int[] GetNppVersion()
{
    int version = Win32.SendMessage(PluginBase.nppData._nppHandle, (uint)NppMsg.NPPM_GETNPPVERSION, 0, 0).ToInt32();
    int major = version >> 16;
    int minor = Math.DivRem(version & 0xffff, 10, out int bugfix);
    return new int[] { major, minor, bugfix };
}
```

**Pull request closed without merging**